_benchmark/creating-a-custom-workload.md

```diff
@@ -0,0 +1,374 @@
+
+```
+head -n 1000 <index>-documents.json > <index>-documents-1k.json
+```
+
+Then, run `opensearch-benchmark execute-test` with the option `--test-mode`. Test mode runs a quick version of the workload test.
+
+```
+opensearch-benchmark execute_test \
+--pipeline="benchmark-only"  \
+--workload-path="<PATH OUTPUTTED IN THE OUTPUT OF THE CREATE-WORKLOAD COMMAND>" \
+--target-host="<CLUSTER ENDPOINT>" \
+--client-options"basic_auth_user:'<USERNAME>',basic_auth_password:'<PASSWORD>'" \
+--test-mode
+```
+
+### Adding variance to test procedures
+
+After using your custom workload several times, you might want to use the same workload but perform the workload's operations in a different order. Instead of creating a new workload or reorganizing the procedures directly, you can provide test procedures to vary workload operations. 
+
+To add variance, go to your `workload.json` file and replace the `schedule` section with a `test_procedures` array, as shown in the following example. Each item in the array contains the following:
+
+- `name`: The name of the test procedure.
+- `default`: When set to `true`, the default `test-procedure` is used even when no operations are specified. 
+- `schedule`: All the operations the test procedure will run.
+
+
+```json
+"test_procedures": [
+    {
+      "name": "index-and-query",
+      "default": true,
+      "schedule": [
+        {
+          "operation": {
+            "operation-type": "delete-index"
+          }
+        },
+        {
+          "operation": {
+            "operation-type": "create-index"
+          }
+        },
+        {
+          "operation": {
+            "operation-type": "cluster-health",
+            "request-params": {
+              "wait_for_status": "green"
+            },
+            "retry-until-success": true
+          }
+        },
+        {
+          "operation": {
+            "operation-type": "bulk",
+            "bulk-size": 5000
+          },
+          "warmup-time-period": 120,
+          "clients": 8
+        },
+        {
+          "operation": {
+            "operation-type": "force-merge"
+          }
+        },
+        {
+          "operation": {
+            "name": "query-match-all",
+            "operation-type": "search",
+            "body": {
+              "query": {
+                "match_all": {}
+              }
+            }
+          },
+          "clients": 8,
+          "warmup-iterations": 1000,
+          "iterations": 1000,
+          "target-throughput": 100
+        }
+      ]
+    }
+  ]
+}
+```
+
+### Separate operations and test procedures
+
+If you want to make your `workload.json` file more readable, you can separate your operations and test procedures into different directories and reference the path to each in `workload.json`. To separate operations and procedures, perform the following steps:
+
+1. Add all test procedures to a file named `index-and-query.json`. 
```

Comment:
`index-and-query.json` can be named anything. I just used this as an example since the example JSON above was named `index-and-query`. To make it less confusing for readers, we can add "Since the example test_procedure above has a `name` of `index-and-query`, we will add all test procedures to a file named `index-and-query`."

Reply:
Are you ok with streamlined approach I took here, or do you think it would be easier to work through this section with the example files?

Reply:
It's fine with the streamlined approach. We can see what others think after it's been released and always improve upon it.